### PR TITLE
Flashback: use castKeyword

### PR DESCRIPTION
**Pull request by Hanmac**
This makes Flashback use castKeyword

meaning if a spell has multiple instances of the keyword, the Exile Replacement should only activate once

Also casting a card with Flashback doesn't add a second instance onto it if its intrinsic

Keyword instances now have HostCard and Intrinsic available (its mostly knowing when the SpellAbility is created in GameActionUtil for Split Cards)

### CODE REVIEW

**Review by Agetian — DISMISSED** [review]

File: forge-game/src/main/java/forge/game/GameAction.java [hunk 1]
```diff
@@ -619,7 +619,7 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
         checkStaticAbilities();
 
         // 400.7g try adding keyword back into card if it doesn't already have it
-        if (zoneTo.is(ZoneType.Stack) && cause != null && cause.isSpell() && c.equals(cause.getHostCard())) {
+        if (zoneTo.is(ZoneType.Stack) && cause != null && cause.isSpell() && !cause.isIntrinsic() && c.equals(cause.getHostCard())) {
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/GameAction.java:622]: Sorry, still found a way to break it:

When it gains two flashback (e.g. Flicker Snapcaster after resolving) and then cast it via the _first_ altcost it won't be exiled

Works with any spell

Comment (Hanmac) [reply forge-game/src/main/java/forge/game/GameAction.java:622]: you don't need to flicker it, just cast a second Snapcaster

i just don't understand it why it gets two times the keywords when moving to stack, i need to debug it more

Comment (Hanmac) [reply forge-game/src/main/java/forge/game/GameAction.java:622]: @tool4ever found the reason: getReplacementEffects returns an FCollection meaning it does filter duplicates

meaning Line 270+ with `copied.setChangedCardKeywords(c.getChangedCardKeywords());` already added the Flashkeyword to stack, but as a copy not as direct value (keywords doesn't have ID yet)
i made it so it doesn't copy when moving to stack


Can we try to get back why this was done?
a624cf565a6adf3fd12d968311acecd99cbdd4c9

Comment (tool4ever) [reply forge-game/src/main/java/forge/game/GameAction.java:622]: From the timing I'd say we had the same issue then :P

image

Comment (Hanmac) [reply forge-game/src/main/java/forge/game/GameAction.java:610]: the Old Code for Flashback isn't needed anymore now that the new code especially can handle the CastKeyword

File: forge-game/src/main/java/forge/game/GameAction.java [hunk 2]
```diff
@@ -265,25 +265,13 @@ private Card changeZone(final Zone zoneFrom, Zone zoneTo, final Card c, Integer
...
-                copied.setChangedCardTypesCharacterDefining(c.getChangedCardTypesCharacterDefiningTable());
-                copied.setChangedCardNames(c.getChangedCardNames());
-                copied.setChangedCardTraits(c.getChangedCardTraits());
-                copied.setDrawnThisTurn(c.getDrawnThisTurn());
-
-                copied.copyChangedTextFrom(c);
-
-                // clean up changes that come from its own static abilities
-                copied.cleanupCopiedChangesFrom(c);
+                // try not to copy changed stats when moving to stack
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/GameAction.java:268]: Seems all Bestowed now disappear from Game on resolve 💩

Comment (Hanmac) [reply forge-game/src/main/java/forge/game/GameAction.java:268]: i moved setSplitStateToPlayAbility up a bit, so i could skip a extra call to `setSplitStateToPlayAbility`

**Review by tool4ever — DISMISSED** [review]
Couldn't break it again 👍

**Comment by Hanmac** [comment]
i found a way to refactor the Riot checks without making it crash into StackOverflow

**Review by tool4ever — APPROVED** [review]

**Pull request merged**